src/toast/scripts/toast_map_stats.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2025-2025 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+"""
+This script loads toast mapmaker outputs and computes various quantities that
+are useful for quality assurance, data cuts, and weighted coadds.
+"""
+
+import argparse
+import os
+
+import numpy as np
+import healpy as hp
+import astropy.io
+from astropy import units as u
+from astropy.table import Column, QTable
+from matplotlib import pyplot as plt
+
+import toast
+from toast.utils import stdouterr_redirected
+from toast.pixels_io_healpix import read_healpix
+from toast.pixels_io_wcs import read_wcs
+from toast.scripts.toast_plot_healpix import main as healpix_main
+from toast.scripts.toast_plot_wcs import main as wcs_main
+
+
+def plot_maps(args, remaining, out_dir):
+    # Build up the arguments to pass to the plotting routine
+    plot_opts = [
+        "--out_dir",
+        out_dir,
+        "--map_file",
+        args.map_file,
+        "--hit_file",
+        args.hits_file,
+        "--cmap",
+        "bwr",
+    ]
+    plot_opts.extend(remaining)
+    try:
+        healpix_main(opts=plot_opts)
+    except Exception as hpe:
+        try:
+            wcs_main(opts=plot_opts)
+        except Exception as wpe:
+            msg = "Failed to plot maps as either healpix or WCS"
+            raise RuntimeError(msg)
+
+
+def compute_spectra(args, remaining, out_dir):
+    try:
+        hits = read_healpix(args.hits_file, nest=False)
+        use_healpix = True
+    except Exception as ehpx:
+        try:
+            hits = read_wcs(args.hits_file)
+            use_healpix = False
+        except Exception as ewcs:
+            msg = "Cannot read hits file as either healpix or WCS format"
+            raise RuntimeError(msg)
+
+    if not use_healpix:
+        raise NotImplementedError("WCS processing not yet implemented")
+
+    good = hits > 3
+    bad = np.logical_not(good)
+    nside = hp.get_nside(hits)
+    data_map, data_header = read_healpix(args.map_file, field=None, nest=False, h=True)
+
+    if "UNITS" in data_header:
+        map_units = u.Unit(data_header["UNITS"])
+    else:
+        map_units = u.K
+    spec_units = map_units**2
+
+    # We just use the intensity covariance for weighting
+    invcov = read_healpix(args.invcov_file, field=(0,), nest=False)
+
+    fsky = np.mean(invcov**2)
+    weighted_map = data_map * invcov
+    weighted_map[:, bad] = 0.0
```

Comment:
Technically, you should set `invcov[bad] = 0` before measuring `fsky`. The effect is obviosly tiny, until a corner case comes around.

Reply:
Done.